### PR TITLE
[Github] Adding templates for ISSUE & PRs and environment script

### PR DESCRIPTION
**Pull request by Sajjon**
Partial fix of #46 + some extra goodies. This PR is a version of https://github.com/apple/swift-crypto/pull/21 updated for this repo, copy pasting my description from that PR in here (we developers are a lazy bunch right?)

Upgrading ISSUE_TEMPLATE from single markdown to 4 option selection: Bug, Regression, Question, Feature Request

### Motivation:

Having this template where when user presses `NEW ISSUE` button she has to think about what she's doing might lead to better issues submitted. 

### Modifications:

1. Adding a `.github` directory containing four different issues types.
2. Added new script `./Scripts/environment.sh` which puts to together a summary of users environment and copies to pasteboard (asking first). The scripts output looks like this:

```bash
Swift version: Apple Swift version 5.2 (swiftlang-1103.0.22 clang-1103.0.22)
Target: x86_64-apple-darwin19.3.0
Unix version: Darwin Sajjode.local 19.3.0 Darwin Kernel Version 19.3.0: Thu Jan  9 20:58:23 PST 2020
 root:xnu-6153.81.5~1/RELEASE_X86_64 x86_64

macOS version: 10.15.3
Xcode-select path: '/Applications/Xcode.app/Contents/Developer
Xcode: Xcode 11.4 (Build version 11N111s)

```

#### New-Issue alternatives
new_issue_options

#### Bug report
With having pasted output generated from `./Scripts/environment.sh`
bug_report

#### Feature request

feature_request

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/FEATURE_REQUEST.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+name: 💡 Feature Request
```
Comment (Sajjon) [inline .github/ISSUE_TEMPLATE/FEATURE_REQUEST.md]: TBD: Need to agree on (potential) emoji and the name

(this is what is used in [Swift Crypto](https://github.com/apple/swift-crypto/blob/master/.github/ISSUE_TEMPLATE/FEATURE_REQUEST.md))

File: .github/ISSUE_TEMPLATE/QUESTION.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+---
+name: ❓ Support Question
```
Comment (Sajjon) [inline .github/ISSUE_TEMPLATE/QUESTION.md]: TBD: Need to agree on (potential) emoji and the name

(this is what is used in [Swift Crypto](https://github.com/apple/swift-crypto/blob/master/.github/ISSUE_TEMPLATE/QUESTION.md))

File: .github/ISSUE_TEMPLATE/REGRESSION.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+---
+name: 😱 Regression
```
Comment (Sajjon) [inline .github/ISSUE_TEMPLATE/REGRESSION.md]: TBD: Need to agree on (potential) emoji and the name

(this is what is used in [Swift Crypto](https://github.com/apple/swift-crypto/blob/master/.github/ISSUE_TEMPLATE/REGRESSION.md))

File: .github/ISSUE_TEMPLATE/BUG_REPORT.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
+---
+name: 🐛 Bug Report
```
Comment (Sajjon) [inline .github/ISSUE_TEMPLATE/BUG_REPORT.md]: TBD: Need to agree on (potential) emoji and the name

(this is what is used in [Swift Crypto](https://github.com/apple/swift-crypto/blob/master/.github/ISSUE_TEMPLATE/BUG_REPORT.md))

**Comment by natecook1000** [comment]
Thanks for putting together this PR, @Sajjon! Just wanted to note that we're looking at these, but need to think a little more about what kind of guidance we want in each of these places.

One Q: Is it possible to have multiple PR templates?

**Comment by Sajjon** [comment]
Yes [you can create multiple PR templates too](
https://help.github.com/en/github/building-a-strong-community/creating-a-pull-request-template-for-your-repository
)

So maybe one PR template for bug fixes, one for general improvement (typos,
code structure/refactoring), one for new features, and possibly one meta,
for github/gir stuff?

**Comment by Sajjon** [comment]
@natecook1000 Yes, [you can create multiple PR templates too](https://help.github.com/en/github/building-a-strong-community/creating-a-pull-request-template-for-your-repository)

So maybe:  
* PR template for bug fixes  
* General improvement (typos, code structure/refactoring)  
* New features  
* Meta, for Github/git stuff?

**Comment by natecook1000** [comment]
@Sajjon I'd like to keep things a bit more streamlined, and had too much feedback to give here, so I added a commit. This chops it down to two issue templates, one for bugs and one for feature requests, plus a link to the discussion forums for questions, etc. Pull requests don't seem to provide the nice "chooser" interface, so I made the default one just for regular changes/fixes and added a more in-depth template for feature additions.

What do you think of these to start with? Thanks again for getting this ball rolling!

**Comment by natecook1000** [comment]
On second thought, I'm going to merge these and we can iterate on them in place.

**Pull request merged**